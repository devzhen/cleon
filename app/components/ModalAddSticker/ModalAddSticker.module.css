.background {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 9;
}

.container {
  width: 421px;
  height: 428px;
  padding: 18px 14px 14px 14px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: var(--color-modal-background);
  border: 1px solid white;
  box-shadow: 0 4px 18px rgba(0, 0, 0, 0.18);
  z-index: 10;
}

.container div[data-role="image-wrapper"] {
  display: flex;
  justify-content: flex-end;
}

.container div[data-role="image-wrapper"] img {
  width: 52px;
  height: 52px;
  cursor: pointer;
}

.container div[data-role="image-wrapper"] img:nth-child(1) {
  margin-right: 10px;
}

.container textarea{
  width: 393px;
  height: 290px;
  background-color: var(--color-text-area-background);
  border: 1px solid var(--color-text-area-border);
  color:  var(--color-text-area-color);
  font-size: 18px;
  padding: 13px;
  line-height: 1.2;
  resize: none;
  overflow: hidden;
  outline: none;
}

.container div[data-role="blur-wrapper"] {
  opacity: 0.5;
  pointer-events: none;
  cursor: not-allowed;
}
