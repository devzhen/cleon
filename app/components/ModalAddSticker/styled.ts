import styled from "styled-components";

export const Background = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 9;
`;

export const Container = styled.div`
  width: 421px;
  height: 428px;
  padding: 18px 14px 14px 14px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: #fff8c8;
  border: 1px solid white;
  box-shadow: 0 4px 18px rgba(0, 0, 0, 0.18);
  z-index: 10;
`;

export const BlurWrapper = styled.div<{ $isDisabled: boolean}>`
  ${(props) => props.$isDisabled && `opacity: 0.5`};
  ${(props) => props.$isDisabled && `pointer-events: none`};
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: flex-end;

  img {
    width: 52px;
    height: 52px;
    cursor: pointer;
  }

  img:nth-child(1) {
    margin-right: 10px;
  }
`;

export const TextArea = styled.textarea`
  width: 393px;
  height: 290px;
  background-color: #fffde6;
  border: 1px solid #dfe2e7;
  color: #2e2e2e;
  font-size: 18px;
  padding: 13px;
  line-height: 1.2;
  resize: none;
  overflow: hidden;
  outline: none;
`;