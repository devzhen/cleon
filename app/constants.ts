export const DEFAULT_COORDS = {
  top: 100,
  left: 220,
} as const;

export const BOARD_TYPE = {
  left: 'left',
  right: 'right',
} as const;