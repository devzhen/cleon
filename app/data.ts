import { v4 as uuid } from "uuid";

export const stickers = [
  {
    id: uuid(),
    top: 110,
    left: 230,
    createdAt: new Date().toISOString(),
    text: ''
  }
];
