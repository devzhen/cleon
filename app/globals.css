:root {
  --color-active-board: #eb6529;
  --color-not-active-board: #bfc3ca;
  --color-main-background: #f0cc46;
  --color-board-background: #e9ecf1;
  --color-header-right-side: #7b7b7b;
  --color-modal-background: #fff8c8;
  --color-text-area-background: #fffde6;
  --color-text-area-border: #dfe2e7;
  --color-text-area-color: #2e2e2e;
  --color-date-color: #979797;
}

html,
body {
  max-width: 100vw;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.main {
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;
}

.boardActive {
  border-color: var(--color-active-board) !important;
}

.boardNotActive {
  border-color: var(--color-not-active-board) !important;
}
