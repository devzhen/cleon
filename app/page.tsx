"use client";

import styled from "styled-components";

import Header from "./components/Header";
import LeftSide from "./components/LeftSide";

export default function Home() {
  return (
    <Main>
      <header>
        <Header />
      </header>
      <LeftSide />
    </Main>
  );
}

const Main = styled.main`
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;

  header {
    grid-column: 2 span;
    height: 66px;
    width: 100%;
  }
`;
