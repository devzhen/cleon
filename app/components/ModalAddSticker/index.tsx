import Image from "next/image";
import { ChangeEvent, useState } from "react";

import { BOARD_TYPE } from "@/app/constants";

import {
  Background,
  Container,
  TextArea,
  ImageWrapper,
  BlurWrapper,
} from "./styled";
import { Sticker } from "@/app/data";

type ModalAddStickerProps = {
  board: (typeof BOARD_TYPE)[keyof typeof BOARD_TYPE];
  closeModal: () => void;
  createSticker: ({
    text,
    board,
    id,
  }: {
    text: string;
    board: (typeof BOARD_TYPE)[keyof typeof BOARD_TYPE];
    id: string | undefined;
  }) => void;
  sticker: Sticker | null;
};

const ModalAddSticker = (props: ModalAddStickerProps) => {
  const { closeModal, createSticker, board, sticker } = props;

  const [text, setText] = useState(sticker?.text || "");

  const addStickerHandler = () => {
    createSticker({ text, board, id: sticker?.id });
  };

  const onChangeHandler = (e: ChangeEvent<HTMLTextAreaElement>) => {
    const text = e.target.value;

    setText(text);
  };

  return (
    <Background>
      <Container>
        <span>Entrez votre nom</span>
        <TextArea onChange={onChangeHandler} value={text} />
        <ImageWrapper>
          <Image
            src="/cross.png"
            width={52}
            height={52}
            alt="Picture of the author"
            draggable={false}
            onClick={closeModal}
          />
          <BlurWrapper $isDisabled={!text}>
            <Image
              src="/check.png"
              width={52}
              height={52}
              alt="Picture of the author"
              draggable={false}
              onClick={addStickerHandler}
            />
          </BlurWrapper>
        </ImageWrapper>
      </Container>
    </Background>
  );
};

export default ModalAddSticker;
