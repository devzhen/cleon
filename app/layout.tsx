import type { Metadata } from "next";
import { Roboto } from "next/font/google";

import "./globals.css";

import StyledComponentsRegistry from "../lib/registry";

const roboto = Roboto({
  weight: ["400", "700"],
  style: ["normal"],
  subsets: ["latin"],
  display: "swap",
});

export const metadata: Metadata = {
  title: "Cleon",
  description: "Cleon",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={roboto.className}>
        <StyledComponentsRegistry>{children}</StyledComponentsRegistry>
      </body>
    </html>
  );
}
