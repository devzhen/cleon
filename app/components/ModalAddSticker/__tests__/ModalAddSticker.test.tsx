import { render } from '@testing-library/react';

import { BOARD_TYPE } from '@/app/constants';
import { firstId, stickers } from '@/app/data';

import ModalAddSticker from '..';

describe('ModalAddSticker tests', () => {
  const props = {
    board: BOARD_TYPE.left,
    closeModal: jest.fn(),
    createSticker: jest.fn(),
    sticker: stickers[firstId],
  };
  it('matches the snapshot', () => {
    const { container } = render(<ModalAddSticker {...props} />);

    expect(container).toMatchSnapshot();
  });
});
