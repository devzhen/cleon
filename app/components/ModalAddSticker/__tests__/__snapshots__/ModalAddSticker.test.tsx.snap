// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAddSticker tests matches the snapshot 1`] = `
<div>
  <div
    class="background modal-bg"
  >
    <div
      class="container"
    >
      <span>
        Entrez votre nom
      </span>
      <textarea>
        Draggable element 1
      </textarea>
      <div
        data-role="image-wrapper"
      >
        <img
          alt="Picture of the author"
          data-nimg="1"
          decoding="async"
          draggable="false"
          height="52"
          loading="lazy"
          src="/_next/image?url=%2Fcross.png&w=128&q=75"
          srcset="/_next/image?url=%2Fcross.png&w=64&q=75 1x, /_next/image?url=%2Fcross.png&w=128&q=75 2x"
          style="color: transparent;"
          width="52"
        />
        <div>
          <img
            alt="Picture of the author"
            data-nimg="1"
            decoding="async"
            draggable="false"
            height="52"
            loading="lazy"
            src="/_next/image?url=%2Fcheck.png&w=128&q=75"
            srcset="/_next/image?url=%2Fcheck.png&w=64&q=75 1x, /_next/image?url=%2Fcheck.png&w=128&q=75 2x"
            style="color: transparent;"
            width="52"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
